### PR TITLE
Backports mech heat fixes from bay

### PR DESCRIPTION
**Pull request by CrimsonShrike**
Fixes solar storms annihilating them and them being deleted by fires compared to other mobs.

### CODE REVIEW

File: code/modules/mechs/mech_life.dm [hunk 1]
```diff
@@ -67,13 +67,18 @@
 /mob/living/exosuit/handle_environment(var/datum/gas_mixture/environment)
 	if(!environment) return
 	//Mechs and vehicles in general can be assumed to just tend to whatever ambient temperature
-	if(abs(environment.temperature - bodytemperature) > 10 )
-		bodytemperature += ((environment.temperature - bodytemperature) / 3)
-
-	if(environment.temperature > material.melting_point * 1.25 ) //A bit higher because I like to assume there's a difference between a mech and a wall
```
Comment (CrimsonShrike) [inline code/modules/mechs/mech_life.dm:73]: Main issue was that bodytemp was ignored, also damage ticks could be in the many hundreds.

File: code/modules/events/solar_storm.dm [hunk 2]
```diff
@@ -39,7 +41,7 @@
 			continue
 
 		//Apply some heat or burn damage from the sun.
-		if(istype(L, /mob/living/carbon/human))
+		if(istype(L, /mob/living/carbon/human) || istype(L, /mob/living/exosuit))
```
Comment (MistakeNot4892) [inline code/modules/events/solar_storm.dm]: This should be some `AdjustBodyTemp()` proc that humans and exosuits can override.

File: code/modules/events/solar_storm.dm [hunk 3]
```diff
@@ -31,6 +31,8 @@
 /datum/event/solar_storm/proc/radiate()
 	// Note: Too complicated to be worth trying to use the radiation system for this.  Its only in space anyway, so we make an exception in this case.
 	for(var/mob/living/L in GLOB.living_mob_list_)
+		if(istype(L.loc, /mob/living/exosuit)) //Todo, generalize this further like I did for rads if other things need to block or mitigate heat - CrimsonShrike
```
Comment (MistakeNot4892) [inline code/modules/events/solar_storm.dm]: This is the kind of pattern that helped make old mech code absolute shit to deal with. An `ATOM_FLAG_SHIELD_CONTENTS` set on mechs would make more sense.

**Review by MistakeNot4892 — CHANGES_REQUESTED** [review]

File: code/modules/events/solar_storm.dm (hunk 2 above)
Comment (CrimsonShrike) [reply code/modules/events/solar_storm.dm]: That naming was used by a proc that I dont quite understand, but done.

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**